fabric-network/test/impl/query/queryagent.js

```diff
@@ -0,0 +1,163 @@
+				connection: {name: 'peer1'}
+			},
+			{
+				response: {status: 200, payload: Buffer.from('peer2')},
+				connection: {name: 'peer2'}
+			}
+		]
+	};
+	const validResonse = {
+		responses: [
+			{
+				response: {status: 200, payload: Buffer.from('peer1')},
+				connection: {name: 'peer1'}
+			}
+		]
+	};
+	const errorResonses = {
+		errors: [
+			error1,
+			error2
+		]
+	};
+	const errorResonse = {
+		errors: [
+			error1
+		]
+	};
+
+
+	beforeEach(() => {
+		query = sinon.createStubInstance(Query);
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	describe('#constructor', () => {
+
+		it('no options', async () => {
+			const queryAgent = new QueryAgent({});
+			expect(queryAgent).to.exist;
+		});
+		it('with options', async () => {
+			const queryAgent = new QueryAgent({}, {something: 33});
+			expect(queryAgent).to.exist;
+		});
+		it('with timeout options', async () => {
+			const queryAgent = new QueryAgent({}, {timeout: 3});
+			expect(queryAgent.requestTimeout).to.equal(3000);
+		});
+	});
+
+	describe('#send', () => {
+		let queryAgent;
+
+		beforeEach(() => {
+			queryAgent = new QueryAgent(query);
+		});
+
+		it('queries a single peer', async () => {
+			await queryAgent.send([peer1]);
+
+			const expected = {
+				targets: [peer1],
+				requestTimeout: 3000
+			};
+			sinon.assert.calledWithMatch(query.send, expected);
+		});
+
+		it('queries multiple peers', async () => {
+			const peers = [peer1, peer2];
+
+			await queryAgent.send(peers);
+
+			const expected = {
+				targets: peers,
+				requestTimeout: 3000
+			};
+			sinon.assert.calledWithMatch(query.send, expected);
+		});
+
+		it('returns query results from one peer', async () => {
+			query.send.resolves(validResonse);
+
+			const result = await queryAgent.send([peer1]);
+
+			const expected = {
+				peer1: {status: 200, payload: Buffer.from('peer1'), message: undefined}
+			};
+			expect(result).to.own.deep.equal(expected);
+		});
+		it('returns query error result from one peer', async () => {
+			query.send.resolves(errorResonse);
+
+			const result = await queryAgent.send([peer1]);
+
+			const expected = {
+				peer1: error1
+			};
+			expect(result).to.own.deep.equal(expected);
+		});
+		it('returns query results from one peers', async () => {
+			query.send.resolves(validResonses);
+
+			const result = await queryAgent.send([peer1, peer2]);
+
+			const expected = {
+				peer1: {status: 200, payload: Buffer.from('peer1'), message: undefined},
+				peer2: {status: 200, payload: Buffer.from('peer2'), message: undefined}
+			};
+			expect(result).to.own.deep.equal(expected);
+		});
+		it('returns query error result from two peers', async () => {
+			query.send.resolves(errorResonses);
+
+			const result = await queryAgent.send([peer1, peer2]);
+
+			const expected = {
+				peer1: error1,
+				peer2: error2
+			};
+			expect(result).to.own.deep.equal(expected);
+		});
+		it('returns no result error result from two peers', async () => {
```

Comment:
This implies that some results might be missing? If so, that sounds very tricky to handle and should be covered by a test.

Reply:
fixed